src/runtime/vm.rs

```diff
@@ -0,0 +1,387 @@
+use crate::runtime::{
+    backing::{ImportBacking, LocalBacking},
+    sig_registry::SigRegistry,
+};
+use std::{mem, ptr};
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct Ctx {
+    /// A pointer to an array of locally-defined memories, indexed by `MemoryIndex`.
+    pub memories: *mut LocalMemory,
+
+    /// A pointer to an array of locally-defined tables, indexed by `TableIndex`.
+    pub tables: *mut LocalTable,
+
+    /// A pointer to an array of locally-defined globals, indexed by `GlobalIndex`.
+    pub globals: *mut LocalGlobal,
+
+    /// A pointer to an array of imported memories, indexed by `MemoryIndex,
+    pub imported_memories: *mut ImportedMemory,
+
+    /// A pointer to an array of imported tables, indexed by `TableIndex`.
+    pub imported_tables: *mut ImportedTable,
+
+    /// A pointer to an array of imported globals, indexed by `GlobalIndex`.
+    pub imported_globals: *mut ImportedGlobal,
+
+    /// A pointer to an array of imported functions, indexed by `FuncIndex`.
+    pub imported_funcs: *mut ImportedFunc,
+
+    /// Signature identifiers for signature-checked indirect calls.
+    pub signatures: *const SigId,
+
+    /// The parent instance.
+    pub local_backing: *mut LocalBacking,
```

Comment:
Since `local_backing` is a parent instance. Could we rename it so is easier to follow?

Maybe from `LocalBacking` to `InstanceBacking` (or `ParentInstance`, or maybe there are better names?)

Reply:
I think the comment isn't really accurate actually. It makes sense to call it local_backing, since it owns the memory storing the local portions of the vmctx.

Reply:
Was a bit hard for me to know what `LocalBacking` was for until I dig more into the implementation. But we can proceed with `LocalBacking` you if you think it's the best naming

Reply:
Yeah, I think local backing is the most accurate name for it right now.